src/linesearch/line_functions.jl

```diff
@@ -13,11 +13,27 @@ represents the function ϕ : R → R defined by
     ϕ(t) := f(x + td).
 """
 type LineFunction <: AbstractNLPModel
```

Comment:
I find it hard to guess what `redirect!` does but I can't really think of a better name...